tools/ci/profile_downloader.py

```
@@ -0,0 +1,157 @@
+            raise RuntimeError(f"source distribution digest mismatch for {requirement}")
+        temporary.replace(target)
+    finally:
+        temporary.unlink(missing_ok=True)
+
+
+def download_packages(destination: Path, requirements: list[str]) -> None:
+    if not requirements or len(requirements) > MAX_PACKAGE_COUNT:
+        raise ValueError(f"profile package count must be between 1 and {MAX_PACKAGE_COUNT}")
+    destination.mkdir(parents=True, exist_ok=True)
+    for requirement in requirements:
+        result = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)